test/dist_model_parallel_test.py

```diff
@@ -0,0 +1,138 @@
+import torch
+import torch.nn as nn
+import torch.distributed.autograd as dist_autograd
+import torch.distributed.rpc as rpc
+# from torch.distributed.optim import DistributedOptimizer
+# from torch.distributed.rpc import RRef
+
+from dist_utils import INIT_METHOD_TEMPLATE, dist_init, TEST_CONFIG
+
+import unittest
+
+def _call_method(method, obj_rref, *args, **kwargs):
+    return method(obj_rref.local_value(), *args, **kwargs)
+
+
+def _remote_method(method, obj_rref, *args, **kwargs):
+    return rpc.rpc_sync(
+        obj_rref.owner(),
+        _call_method,
+        args=[method, obj_rref] + list(args),
+        kwargs=kwargs
+    )
+
+class RemoteModule(nn.Module):
+    def __init__(self):
+        super(RemoteModule, self).__init__()
+
+    def remote_parameters(self):
+        param_rrefs = []
+        for param in self.parameters():
+            param_rrefs.append(RRef(param))
+        return param_rrefs
+
+def _get_remote_parameters(remote_module_rref):
+    return rpc.rpc_sync(
+        remote_module_rref.owner(),
+        _call_method,
+        args=[RemoteModule.remote_parameters, remote_module_rref]
+    )
+
+class Encoder(RemoteModule):
```

Comment:
nit, should this be called something like `EmbeddingLayer` or `RemoteEmbeddingLayer`? Usually in seq2seq models encoder refers to the RNN that creates the encoded representation, which in this case would be the LSTM.